pandas/io/formats/format.py

```diff
@@ -565,315 +569,409 @@ def __init__(
-                )
-                # truncate formatter
-                if isinstance(self.formatters, (list, tuple)):
-                    truncate_fmt = self.formatters
-                    self.formatters = [
-                        *truncate_fmt[:col_num],
-                        *truncate_fmt[-col_num:],
-                    ]
-            self.tr_col_num = col_num
-        if truncate_v:
-            # cast here since if truncate_v is True, max_rows_adj is not None
-            max_rows_adj = cast(int, max_rows_adj)
-            if max_rows_adj == 1:
-                row_num = max_rows
-                frame = frame.iloc[:max_rows, :]
-            else:
-                row_num = max_rows_adj // 2
-                frame = concat((frame.iloc[:row_num, :], frame.iloc[-row_num:, :]))
-            self.tr_row_num = row_num
-        else:
-            self.tr_row_num = None
+        if self.is_truncated_horizontally:
+            self._truncate_horizontally()
 
-        self.tr_frame = frame
-        self.truncate_h = truncate_h
-        self.truncate_v = truncate_v
-        self.is_truncated = bool(self.truncate_h or self.truncate_v)
+        if self.is_truncated_vertically:
+            self._truncate_vertically()
 
-    def _to_str_columns(self) -> List[List[str]]:
-        """
-        Render a DataFrame to a list of columns (as lists of strings).
+    def _truncate_horizontally(self) -> None:
+        """Remove columns, which are not to be displayed and adjust formatters.
+
+        Attributes affected:
+            - tr_frame
+            - formatters
+            - tr_col_num
         """
-        # this method is not used by to_html where self.col_space
-        # could be a string so safe to cast
-        col_space = {k: cast(int, v) for k, v in self.col_space.items()}
+        assert self.max_cols_fitted is not None
+        col_num = self.max_cols_fitted // 2
+        if col_num >= 1:
+            cols_to_keep = [
+                x
+                for x in range(self.frame.shape[1])
+                if x < col_num or x >= len(self.frame.columns) - col_num
+            ]
+            self.tr_frame = self.tr_frame.iloc[:, cols_to_keep]
 
-        frame = self.tr_frame
-        # may include levels names also
+            # truncate formatter
+            if isinstance(self.formatters, (list, tuple)):
+                slicer = itemgetter(*cols_to_keep)
+                self.formatters = slicer(self.formatters)
+        else:
+            col_num = cast(int, self.max_cols)
+            self.tr_frame = self.tr_frame.iloc[:, :col_num]
+        self.tr_col_num = col_num
 
-        str_index = self._get_formatted_index(frame)
+    def _truncate_vertically(self) -> None:
+        """Remove rows, which are not to be displayed.
+
+        Attributes affected:
+            - tr_frame
+            - tr_row_num
+        """
+        assert self.max_rows_fitted is not None
+        row_num = self.max_rows_fitted // 2
```

Comment:
BTW, I find the "max_rows_fitted" naming instead of "max_rows_adjusted" as it was before a bit confusing, as the "fitting" seems to come from "fit on the screen", but the eventual max_rows are not necessarily what fit on the screen, that depends on your settings.

Reply:
OK. Will change in the upcoming PRs.